texar/modules/connectors/connectors.py

```diff
@@ -0,0 +1,866 @@
+        flat_output[i] = torch.squeeze(flat_output[i], 1)
+
+    if isinstance(flat_output_size[0], torch.Size):
+        for (i, shape) in enumerate(flat_output_size):
+            flat_output[i] = torch.reshape(
+                flat_output[i], (-1, ) + shape)
+
+    output = nest.pack_sequence_as(structure=output_size,
+                                   flat_sequence=flat_output)
+    return output
+
+
+class ConstantConnector(ConnectorBase):
+    r"""Creates a constant ``Tensor`` or (nested) ``tuple`` of Tensors that
+    contains a constant value.
+
+    Args:
+        output_size: Size of output **excluding** the batch dimension. For
+            example, set :attr:`output_size` to ``dim`` to generate output of
+            shape ``[batch_size, dim]``.
+            Can be an ``int``, a tuple of ``int``, a ``torch.Size``,
+            or a ``tuple`` of ``torch.Size``.
+            For example, to transform inputs to have decoder state size, set
+            :python:`output_size=decoder.state_size`.
+            If :attr:`output_size` is ``tuple`` ```(1, 2, 3)```, then the
+            output structure will be
+            ``([batch_size * 1], [batch_size * 2], [batch_size * 3])``.
+            If :attr:`output_size` is ``torch.Size([1, 2, 3])``, then the
+            output structure will be ``[batch_size, 1, 2, 3]``.
+        hparams (dict, optional): Hyperparameters. Missing
+            hyperparamerter will be set to default values. See
+            :meth:`default_hparams` for the hyperparameter sturcture and
+            default values.
+    This connector does not have trainable parameters.
+
+    Example:
+
+        .. code-block:: python
+
+            state_size = (1, 2, 3)
+            connector = ConstantConnector(state_size, hparams={"value": 1.})
+            one_state = connector(batch_size=64)
+            # `one_state` structure: (Tensor_1, Tensor_2, Tensor_3),
+            # Tensor_1.size() == torch.Size([64, 1])
+            # Tensor_2.size() == torch.Size([64, 2])
+            # Tensor_3.size() == torch.Size([64, 3])
+            # Tensors are filled with 1.0.
+            size = torch.Size([1, 2, 3])
+            connector_size = ConstantConnector(size, hparams={"value": 2.})
+            size_state = connector_size_0(batch_size=64)
+            # `size_state` structure: Tensor with size [64, 1, 2, 3].
+            # Tensor is filled with 2.0.
+
+    """
+    def __init__(self,
+                 output_size: OutputSize,
+                 hparams: Optional[HParams] = None):
+
+        super().__init__(output_size, hparams)
+
+        self.value = self.hparams.value
+
+    @staticmethod
+    def default_hparams() -> dict:
+        r"""Returns a dictionary of hyperparameters with default values.
+
+        .. code-block:: python
+
+            {
+                "value": 0.,
+                "name": "constant_connector"
+            }
+
+        Here:
+
+        "value" : float
```

Comment:
The correct format for the hparams docstring is now:
```
`"value"`: float
```
The hparam name should be wrapped in backticks(`` ` ``), and there should be no spaces between the name and the colon (`:`).